src/client/array/dc_array.c

```diff
@@ -1091,6 +1091,16 @@ set_short_read_cb(tse_task_t *task, void *data)
 	while (current) {
 		daos_size_t len = 0, recs;
 		int i;
+		d_iov_t sg_iov;
+
+		if (current->user_sgl_used) {
+			D_ASSERT(args->sgl->sg_nr == 1);
+			current->sgl.sg_nr = args->sgl->sg_nr;
+			current->sgl.sg_nr_out = args->sgl->sg_nr_out;
+			sg_iov.iov_buf_len = args->sgl->sg_iovs[0].iov_buf_len;
+			sg_iov.iov_len = args->sgl->sg_iovs[0].iov_len;
+			current->sgl.sg_iovs = &sg_iov;
```

Comment:
Just curious is it ok to  leave the local variable pointer in the current? or do you need clean it before exit?

Reply:
i don't think it breaks anything in this case.
in case user sgl is used, we just set the nr and lengths to handle below, not the buffer pointer bec we don't need that in the short-fetch handling. so it's just setting some values in the current sgl, no pointers to anything.